File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cs
public class Solution {
  public int SubarraySum(int[] nums, int k) {
    int counter = 0;
    var dict = new Dictionary<int, int>{[0] = 1};
    int sum = 0;
    foreach(int num in nums) {
      sum += num;
      if (dict.ContainsKey(sum - k)) {
        counter += dict[sum-k];
      }
      dict[sum] = dict.ContainsKey(sum) ? dict[sum] + 1 : 1;
    }
    
    return counter;
  }
}

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.rs
use std::collections::HashMap;

impl Solution {
  pub fn subarray_sum(nums: Vec<i32>, k: i32) -> i32 {
    let mut dict = HashMap::new();
    dict.insert(0, 1);
    let mut sum = 0;
    let mut counter = 0;
    for &v in nums.iter() {
      sum += v;
      let key = sum-k;
      if dict.contains_key(&key) {
        counter += dict[&key];
      }
      dict.insert(sum, if dict.contains_key(&sum) { dict[&sum]+1 } else { 1 });
    }
    counter
  }
}